### ISSUE TITLE
Compile time regression because of llvm inlining

### ISSUE THREAD

**Issue by detrumi** [lalrpop/lalrpop#543]
We're using `lalrpop` in the `chalk` library ([parser code here](https://github.com/rust-lang/chalk/blob/master/chalk-parse/src/parser.lalrpop)). After [upgrading](https://github.com/rust-lang/chalk/commit/39767db5ad1d194d4d1027c4f2e880d7509ce264#diff-42c2060f98a049b854e891a4a7d5c65bR15) `lalrpop` from 0.17 to 0.19, we detected a big compile time regression.

Bisecting led to [this commit](https://github.com/lalrpop/lalrpop/commit/0c69e99). This change increased the `chalk` release build time from 3:26 to 4:55 on my system.

I understand that this is a trade-off between runtime and compile-time, but would it be possible to reduce the impact, or to make this configurable?

**Comment by Marwes** [comment]
So I guess LLVM ends up inlining the `pop_variant` functions a lot more now, which was the intention as it should help reduce the huge number of memcpys lalrpop does. I never expected it to have this huge of a compilation time trade-off though. I created https://github.com/lalrpop/lalrpop/pull/544 which tries to reduce the cost of any potential inlining. Can you try that?

**Comment by detrumi** [comment]
@Marwes Yes, that change does indeed fix the issue, reducing the compile time by more than a minute.

**Comment by detrumi** [comment]
Is this blocked on something? If not, could you get this landed so it gets included in the next lalrpop version?

**Comment by Marwes** [comment]
Released now!
